### ISSUE TITLE
Confused about validation loss calculation

### ISSUE THREAD

**Issue by quickgrid** [brohrer/sharpened-cosine-similarity#3]
Hello, I am unable to understand the testing loss. Why `loss = F.cross_entropy(preds, labels)` is not also used for test loss with test preds?

https://github.com/brohrer/sharpened_cosine_similarity_torch/blob/28c135adfb6062b0e9e29bb906736ceea9c08162/model_cifar10_18_4.py#L152-L163

**Comment by brohrer** [comment]
Hi Asif,

That's a great observation. I have no idea why it's working without that line, but after I add it in I get the same behavior. I'm guessing I got lucky with the object and function references by reusing variable names. It's a little frightening. I'll get that fixed and pushed back out. Thanks for the heads up.

Brandon

------- Original Message -------

**Comment by quickgrid** [comment]
I am really glad it helped. I was also working on validation loop at the time and it caught my eye easily because of that. If you don't mind my suggestion I think the following could probably be simplified into a single `nn.Sequential` defined in init.

https://github.com/brohrer/sharpened_cosine_similarity_torch/blob/28c135adfb6062b0e9e29bb906736ceea9c08162/model_cifar10_18_4.py#L84-L91

**Comment by brohrer** [comment]
That seems like a nice way to streamline the code and make it more readable. If you'd like to open another issue with that suggestion, that will be a good way to track it. Even better, if you would also like to submit a pull request after you make the change and test it, that would be amazing.
